include/picongpu/unitless/simulation.unitless

```diff
@@ -199,6 +199,9 @@ namespace picongpu
 
         static constexpr uint32_t dim()
         {
+            static_assert(
+                simDim == SIMDIM,
```

Comment:
@psychocoderHPC I assume this is just a check for now until fully getting rid of `SIMDIM` in the code, right?

Reply:
we cann ot get rid of the define SIMDIM because SIMDIM is required for include guards